libraries/AP_HAL_SITL/UARTDriver.cpp

```diff
@@ -283,14 +285,10 @@ size_t UARTDriver::_write(const uint8_t *buffer, size_t size)
             if (fabsf(rand_float()) < _sitl->uart_byte_loss_pct.get() * 0.01 * size) {
                 nwrite--;
             }
-            if (nwrite == 0) {
```

Comment:
this was deliberate as it was simulating byte loss on the UART. When we lose bytes on a uart the write() call doesn't know the bytes are lost. We want the caller to think the bytes went out

Reply:
Oh, duh.  My bad.  I believe I've fixed this.

![image](https://github.com/ArduPilot/ardupilot/assets/7077857/9e28756a-fc43-47e4-a556-49a370298c34)